packages/@react-spectrum/dialog/src/DialogTrigger.tsx

```diff
@@ -66,20 +71,20 @@ export function DialogTrigger(props: SpectrumDialogTriggerProps) {
       case 'fullscreen':
       case 'fullscreenTakeover':
         return (
-          <Modal isOpen={isOpen} onClose={onClose} type={type}>
-            {content}
+          <Modal isOpen={isOpen} isDismissable={false} onClose={onClose} type={type}>
+            {typeof content === 'function' ? content(onClose) : content}
           </Modal>
         );
       case 'modal':
         return (
-          <Modal isOpen={isOpen} onClose={onClose}>
-            {content}
+          <Modal isOpen={isOpen} isDismissable={isDismissable} onClose={onClose}>
+            {typeof content === 'function' ? content(onClose) : content}
           </Modal>
         );
       case 'tray':
         return (
           <Tray isOpen={isOpen} onClose={onClose}>
-            {content}
+            {typeof content === 'function' ? content(onClose) : content}
```

Comment:
hmm... thought tray's couldn't have the buttons? I'm fine to just have it all done the same though, seems useful